lib/alice/adapter/supervisor.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Alice.Adapter.Supervisor do
+  @moduledoc """
+  Supervises any Alice.Adapter process that are started.
+  """
+
+  use Supervisor
+
+  def start_link(_) do
+    Supervisor.start_link([
+      worker(Alice.Adapter, [], restart: :transient)
```

Comment:
We may want to consider converting this to a module based supervisor, [per the docs here] (https://hexdocs.pm/elixir/Supervisor.html#module-module-based-supervisors) particularly since this is a `simple_one_for_one` supervisor. This will be important in particular if we want to support hot code-reloading as a deployment strategy (and I think we do). This change seems pretty simple, as it means moving most things to the `init/1` callback rather than `start_link/2`.

Reply:
Since I wasn't doing anything in `init` for now I used the simplified version, but I think you're right. It will leave it more open for change and doesn't add much, if any, extra complexity.